src-srv/utils/hocuspocus.ts

```diff
@@ -0,0 +1,64 @@
+import type { Hocuspocus } from '@hocuspocus/server'
+import { Server } from '@hocuspocus/server'
+import { Logger } from '@hocuspocus/extension-logger'
+import { Redis } from '@hocuspocus/extension-redis'
+import { Database } from '@hocuspocus/extension-database'
+import type { RedisCache } from './RedisCache.ts'
+import type { Repository } from './Repository.ts'
+import { decodeJwt } from 'jose'
+import { initDoc } from './transformations/index.ts'
+
+interface CreateServerOptions {
+  repository: Repository
+  cache: RedisCache
+}
+
+export async function createServer(options: CreateServerOptions): Promise<Hocuspocus> {
+  const { repository, cache } = options
+
+  const server = Server.configure({
+    port: parseInt(process.env.API_PORT),
+    extensions: [
+      new Logger(),
+      new Redis({
+        host: process.env.REDIS_HOST, // FIXME: Should not use env
+        port: parseInt(process.env.REDIS_PORT) // FIXME: SHould not use env
+      }),
+      new Database({
+        fetch: async (data) => {
```

Comment:
I'm getting lint warnings for `async () =>`
```
  Unexpected space before function parentheses. eslint (space-before-function-paren) [28, 21]
```
Which i guess technically is correct. Dont' have a strong opinion about this.
Some more below....

Reply:
Yes, when async is used in front of anon fat arrow func. I have added eslint rule exception for this
("@typescript-eslint/no-confusing-void-expression": ["error", { ignoreArrowShorthand: true }])